api/src/main/java/io/grpc/InternalConfigSelector.java

```diff
@@ -113,7 +137,7 @@ public Builder setCommittedCallback(@Nullable Runnable committedCallback) {
       }
 
       public Result build() {
-        return new Result(config, callOptions, committedCallback);
+        return new Result(Status.OK, config, callOptions, committedCallback);
```

Comment:
`config` and `callOptions` here must not be null

Reply:
Why they must not be `null`? For cases the status is not OK.

Reply:
For cases is OK, they must be not null.

Reply:
Ok, checked. Does this really provide a stronger API guarantee or it only weakens the specification by requiring a stronger precondition?

Reply:
We should specify the `Result` must be either with error status (all other fields are ignored), or with OK status in which case config and callOptions are not null.

Reply:
It's mentioned in `getStatus()`.

Reply:
It's not mentioning "with OK status, callOptions is not null." I'm not going to be nagging on about this. LGTM.